example/lib/main.dart

```diff
@@ -42,88 +52,93 @@ class _MyHomePageState extends State<MyHomePage> {
   UhstSocket? client;
   final TextEditingController _hostIdController = TextEditingController();
 
-  @override
-  void initState() {
-    super.initState();
-  }
-
-  Future<void> initHost() async {
+  Future<void> connectHost() async {
     initUHST();
     host?.disconnect();
     host = uhst?.host();
     host
       ?..onReady(handler: ({required hostId}) {
-        setState(() {
-          hostMessages.add('Host Ready! Using $hostId');
-          print('host is ready!');
-          _hostIdController.text = hostId;
-        });
+        hostMessages.add('Host Ready! Using hostId $hostId');
+        print('host is ready!');
+        _hostIdController.text = hostId;
+        setState(() {});
       })
       ..onException(handler: ({required dynamic exception}) {
-        print('exception received! $exception');
-        setState(() {
-          hostMessages.add(exception.toString());
-        });
+        if (exception is RelayException) {
+          hostMessages.add('disconneted! $exception');
+        } else {
+          hostMessages.add('exception received! $exception');
+        }
+        setState(() {});
+      })
+      ..onClose(handler: ({required hostId}) {
+        hostMessages.add('Host with id $hostId disconnected');
+        setState(() {});
       })
       ..onConnection(handler: ({required uhstSocket}) {
         uhstSocket
           ..onDiagnostic(handler: ({required message}) {
-            setState(() {
-              hostMessages.add(message);
-            });
+            hostMessages.add(message);
+            setState(() {});
           })
           ..onMessage(handler: ({required message}) {
-            setState(() {
-              hostMessages
-                  .add('Host received: $message from ${uhstSocket.remoteId}');
-            });
+            hostMessages
+                .add('Host received: $message from ${uhstSocket.remoteId}');
+            setState(() {});
             host?.broadcastString(message: message);
           })
           ..onOpen(handler: () {
-            setState(() {
-              hostMessages.add('Client ${uhstSocket.remoteId} connected');
-            });
+            hostMessages.add('Client ${uhstSocket.remoteId} connected');
+            setState(() {});
+          })
+          ..onClose(handler: ({required hostId}) {
```

Comment:
why pass hostId here? This method is called when a client disconnects, the hostId is not relevant.

Reply:
same here:
the reason is that as onClose will be emitted to event stream, and in the moment when listener will be called there will be no hostId in socket as it will be closed - and it will be null

Reply:
fixed